/*
 *  Copyright (C) 2020-2024 Embedded AMS B.V. - All Rights Reserved
 *
 *  This file is part of Embedded Proto.
 *
 *  Embedded Proto is open source software: you can redistribute it and/or 
 *  modify it under the terms of the GNU General Public License as published 
 *  by the Free Software Foundation, version 3 of the license.
 *
 *  Embedded Proto  is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with Embedded Proto. If not, see <https://www.gnu.org/licenses/>.
 *
 *  For commercial and closed source application please visit:
 *  <https://EmbeddedProto.com/license/>.
 *
 *  Embedded AMS B.V.
 *  Info:
 *    info at EmbeddedProto dot com
 *
 *  Postal address:
 *    Atoomweg 2
 *    1627 LE, Hoorn
 *    the Netherlands
 */

// This file is used to test the wire formatter class.

syntax = "proto3";

package demo.space;

message message_a 
{
  repeated int32 x   = 1;
  float y   = 2;
  sint64 z  = 3;
}

message message_b 
{
  double u  = 1;
  message_a nested_a = 2;
  int32 v   = 3;
}

message message_c 
{
  message message_d
  {
    repeated uint32 d = 1;
  }

  message_b nested_b = 1;
  message_d nested_d = 2;

  // Deliberate usage before definition to test topsort algorithm.
  message_e.message_g nested_g = 3;
}

message message_e
{
  
  message message_f
  {
    int32 f = 1;
    // Deliberate usage before definition to test topsort algorithm.
    message_g g_in_f = 2; 
  }

  message message_g
  {
    int32 g = 1;
  }

  message_f nested_f = 1;
  message_g nested_g = 2;

}
